File: LeetCode/Easy/0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        vector<double> ret;
        
        queue<TreeNode*> q;
        q.push(root);

        while (!q.empty()) {
            double temp = 0;
            int size = q.size();
            for (int i = 0; i < size; ++i){
                TreeNode* node = q.front();
                q.pop();
                temp += node->val;
                if (node->left) q.push(node->left);
                if (node->right) q.push(node->right);
            }
            ret.push_back(temp / size);
        }
        
        return ret;
    }
};

File: LeetCode/Easy/0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func averageOfLevels(root *TreeNode) []float64 {
    ret := make([]float64, 0)
    q := make([]*TreeNode, 1)
    q[0] = root
    
    for len(q) > 0 {
        size := len(q)
        temp := 0
        
        for i := 0; i < size; i++ {
            temp += q[i].Val
            if q[i].Left != nil { q = append(q, q[i].Left) }
            if q[i].Right != nil { q = append(q, q[i].Right) }
        }
        
        ret = append(ret, float64(temp) / float64(size))
        q = q[size:]
    }
    
    return ret
}